### PR TITLE
Some updates to docs

### PR DESCRIPTION
**Pull request by erlendvollset**
- Improve filter documentation
- Revise quickstart
- Make TOC slightly more granular. time series and geospatial each have 4 sub resources, and sequences has 2, so bringing them up makes more sense IMO.

(TY ChatGPT) 🙏

### CODE REVIEW

File: cognite/client/data_classes/filters.py [hunk 1]
```diff
@@ -319,6 +400,23 @@ def _filter_body(self, camel_case_property: bool) -> list:
...
 @final
 class Range(FilterWithProperty):
+    """Filters results based on a range of values.
+
+    Args:
+        property (PropertyReference): The property to filter on.
+        gt (FilterValue | None): Greater than.
+        gte (FilterValue | None): Greater than or equal to.
+        lt (FilterValue | None): Less than.
+        lte (FilterValue | None): Less than or equal to.
```
Comment (haakonvt) [inline cognite/client/data_classes/filters.py:410]: Just noticed that these type hints using `FilterValue` are wrong/misleading as only a subset is supported (other misleading uses: `Overlaps` and `Prefix`):

> (RangeValue (RangeValue (string) or RangeValue (number) or RangeValue (integer))) or ReferencedPropertyValueV3 (object) (FilterValueRange)

```py
@dataclass
class PropertyReferenceValue:
    property: PropertyReference

@dataclass
class ParameterValue:
    parameter: str

FilterValue: TypeAlias = Union[RawValue, PropertyReferenceValue, ParameterValue]
RawValue: TypeAlias = Union[str, float, bool, Sequence, Mapping[str, Any], Label]
```

File: docs/source/filters.rst [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+^^^^^^^^
+The `MatchAll` filter combines multiple sub-filters using a logical AND operation.
+
+.. autoclass:: cognite.client.data_classes.filters.MatchAll
+    :members:
+    :member-order: bysource
+
+Nested
+^^^^^^
+The `Nested` filter applies a filter to a nested property.
```
Comment (haakonvt) [inline docs/source/filters.rst:182]: A short explanation of what "nested" or "nested property" means would be great!

File: docs/source/filters.rst [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+^^^^^^^
+The `HasData` filter checks if an item has data for a given property.
+
+.. autoclass:: cognite.client.data_classes.filters.HasData
+    :members:
+    :member-order: bysource
+
+MatchAll
+^^^^^^^^
+The `MatchAll` filter combines multiple sub-filters using a logical AND operation.
```
Comment (haakonvt) [inline docs/source/filters.rst:174]: Copy paste error from `And` filter it looks like

File: docs/source/index.rst [hunk 4]
```diff
@@ -39,17 +39,23 @@ Contents
...
    extensions_and_optional_dependencies
    identity_and_access_management
-   core_data_model
+   data_modeling
+   assets
+   events
+   files
+   time_series
+   sequences
+   geospatial
```
Comment (haakonvt) [inline docs/source/index.rst:48]: Great change 👌 Also love DM at the top!

**Review by haakonvt — COMMENTED** [review]
I really like that you have made a separate top-level page in the docs for filtering. Should allow us to link to it easily from other parts of the documentation without bloating it with (similar-in-nature) examples everywhere 🥇

File: cognite/client/data_classes/filters.py (hunk 1 above)
Comment (erlendvollset) [reply cognite/client/data_classes/filters.py:410]: Yeah, not ideal. Will leave some validation for runtime. But not gonna address that in this PR.

File: cognite/client/data_classes/filters.py [hunk 5]
```diff
@@ -251,16 +251,56 @@ def _filter_body(self, camel_case_property: bool) -> dict[str, Any]:
...
+            >>> from cognite.client.data_classes.filters import Or, Equals, In
+            >>> filter = Or(Equals(("some", "property"), 42), In(("another", "property"), ["a", "b", "c"]))
+    """
+
     _filter_name = "or"
 
 
 @final
 class Not(CompoundFilter):
+    """A filter that negates another filter.
```
[the remark this answers is not in the document]
Comment (erlendvollset) [reply cognite/client/data_classes/filters.py:291]: Not sure about that. It makes it sound like it negates each of the underlying filters recursively, which would be difficult to reason about if it was the case.

**Review by haakonvt — APPROVED** [review]
LGTM 👌

**Pull request merged**